### ISSUE TITLE
Version 7.0.0.0 is binary incompatible with previous versions. An unhandled exception of type 'System.MissingMethodException' occurred in mscorlib.dll 

### ISSUE THREAD

**Issue by DecorteHannes** [Azure/azure-storage-net#297]
Hallo,

Since the following method (among others) has been removed in version 7.0.0.0:

public static TableOperation Retrieve(string partitionKey, string rowkey)

we face the issue that version 7.0.0.0 is not binary compatible with previous versions. It is source level compatible due to availability of method, but not binary.

public static TableOperation Retrieve(string partitionKey, string rowkey, List<string> selectColumns = null)

However in the following use case this is problem:
- We have a binary that is not under our control (actually it is binary Microsoft.WindowsAzure.ResourceStack.Common.Storage.dll) that is compiled against version 4.3.0.0 of Microsoft.WindowsAzure.Storage.
- We need to use that binary
- We are (in our project) on version 7.0.0.0 of Microsoft.WindowsAzure.Storage
- We do binding redirect to version 7.0.0.0 in the config
- There is an exception 
  An unhandled exception of type 'System.MissingMethodException' occurred in mscorlib.dll 
  Additional information: Method not found: Microsoft.WindowsAzure.Storage.Table.TableOperation Microsoft.WindowsAzure.Storage.Table.TableOperation.Retrieve(System.String, System.String).
- Although the Retrieve method on TableOperation in Version 7.0.0.0 is source level compatible, it is not binary compatible.
- We cannot ourselves recompile Microsoft.WindowsAzure.ResourceStack.Common.Storage.dll

Can the binary compatibility be brought back?

**Comment by esummers-msft** [comment]
We follow semantic versioning protocol which guarantees that for minor version updates (e.g. 6.1 to 6.2) we will not allow breaking changes in our library. For major version updates however, like the update to 7.0, we need to make changes which break with previous versions. When upgrading to a new major version, the onus is on each customer to fix any breaks that may have appeared in their code and recompile their applications if needed. Sorry for the inconvenience.

**Comment by DecorteHannes** [comment]
I totally agree. It is up to the owners of libraries that use this assembly to keep up with recent versions.
But since we got a "No we will not bring our Microsoft.WindowsAzure.ResourceStack.Common.Storage.dll - due to lack of resources - to the latest version 7.0.0.0", I tried to get a solution here. 
Just a pity that I have to try to negotiate a solution for a blocking issue between assemblies owned by the same company (Microsoft).
